cfme/tests/cloud/test_provisioning.py

```diff
@@ -125,25 +125,21 @@ def vm_name(request):
     return random_vm_name('prov')
 
 
-@pytest.mark.parametrize('testing_instance', [True, False], ids=["Auto", "Manual"], indirect=True)
-def test_provision_from_template(provider, testing_instance, soft_assert):
-    """ Tests instance provision from template
-
-    Metadata:
-        test_flag: provision
-    """
-    instance, inst_args, image = testing_instance
+def provision_check(provider, instance, inst_args, image):
```

Comment:
**REQUIRED**
After looking more deeply into this module I think this could be a fixture, but with name let's say `provisioned_instance`. Also it should take `testing_instance` fixture as a argument:
```python
@pytest.fixture
def provisioned_instance(provider, testing_instance):
...
```
After that in your tests you can do something like that:
```python
@pytest.mark.parametrize('testing_instance', [True, False], ids=["Auto", "Manual"], indirect=True)
def test_provision_from_template(provider, provisioned_instance, soft_assert):
    """ Tests instance provision from template

    Metadata:
        test_flag: provision
    """
    instance, inst_args, image = testing_instance
    instance = provision_check(provider, instance, inst_args, image)
```
What do you think?

Reply:
Sounds good)  This will perfectly work for provisioning from a template, as for tagging, I updated testing_instance fixture to handle tag option. @quarckster, could you please check it, if it is ok?